storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsShuffleReadHandler.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.storage.handler.impl;
+
+import com.google.common.collect.Lists;
+import com.tencent.rss.common.ShuffleDataResult;
+import com.tencent.rss.common.ShuffleDataSegment;
+import com.tencent.rss.common.ShuffleIndexResult;
+import com.tencent.rss.common.util.RssUtils;
+import com.tencent.rss.storage.util.ShuffleStorageUtils;
+import java.io.IOException;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HdfsShuffleFileReadHandler is a shuffle-specific file read handler, it contains two HdfsFileReader
+ * created by using the index file and its indexed data file.
+ */
+public class HdfsShuffleReadHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(HdfsShuffleReadHandler.class);
+
+  protected final String filePrefix;
+  protected final HdfsFileReader indexReader;
+  protected final HdfsFileReader dataReader;
+  protected final int readBufferSize;
+
+  protected final List<ShuffleDataSegment> shuffleDataSegments = Lists.newArrayList();
+  protected int segmentIndex;
+
+  public HdfsShuffleReadHandler(
+      String fullShufflePath,
+      String filePrefix,
+      int readBufferSize,
+      Configuration conf) throws IOException {
+    this.filePrefix = filePrefix;
```

Comment:
why fullShufflePath and filePrefix should be included?
Full path is enough for HdfsShuffleReadHandler

Reply:
The `fullShuflePath` is actually not appropriate, it is the shuffle's base path without the prefix. Shall we rename `fullShuflePath` to `shuffleBasePath`?